lib/molinillo/dependency_graph.rb

```diff
@@ -248,7 +259,12 @@ def hash
       # dependency graph?
       # @return true iff there is a path following edges within this {#graph}
       def path_to?(other)
-        successors.include?(other) || successors.any? { |v| v.path_to?(other) }
+        walk = proc do |vertex|
+          return true if vertex.equal?(other)
+          vertex.successors.each(&walk)
+          false
+        end
+        walk.call(self)
```

Comment:
I can't figure out what the win of making a proc is here vs just calling the method recursively with `other` as an argument... what am I missing?